src/helpers/utils.js

```diff
@@ -0,0 +1,17 @@
+import { cacheTime } from './queries'
+
+export const isCollectionLoading = col => {
+  if (!col) {
+    console.warn('isCollectionLoading called on falsy value.') // eslint-disable-line no-console
+    return false
+  }
+  return col.fetchStatus === 'loading' || col.fetchStatus === 'pending'
+}
+
+export const isCollectionInCache = col => {
+  if (!col) {
+    console.warn('isCollectionInCache called on falsy value.') // eslint-disable-line no-console
+    return false
+  }
+  return col.lastFetch + cacheTime >= Date.now()
+}
```

Comment:
Why are you doing that? You don't need to check that. CozyClient and its fetch policy will do that for you. In our case, we just need to check if `lastFetch` exists to know if we have datas in the store. Nothing more / less.

Reply:
Also https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/utils.js#L30-L43 ;)

Reply:
You pass a query definition to CozyClient. This QueryDefinition has an `as` and a fetchPolicy. CozyClient, will check if in its own store he has data for this `as`. If there is something, he will check if the `fetchPolicy` is respected or not. If it's `respected` then he doesn't `fetch` he will only read from its store, if not he will `fetch` the data https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/Query.jsx#L106-L117